### PR TITLE
Add generics to the progress bar iteration methods

### PR DESCRIPTION
**Pull request by jdreesen**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| License       | MIT

Adds some generics to the progress bar iteration methods to improve static analysis.

### CODE REVIEW

**Review by lyrixx — APPROVED** [review]
I needed that few days ago! Thanks

File: src/Symfony/Component/Console/Helper/ProgressBar.php [hunk 1]
```diff
@@ -305,7 +305,13 @@ public function maxSecondsBetweenRedraws(float $seconds): void
     /**
      * Returns an iterator that will automatically update the progress bar when iterated.
      *
-     * @param int|null $max Number of steps to complete the bar (0 if indeterminate), if null it will be inferred from $iterable
+     * @template TKey
+     * @template TValue
+     *
+     * @param iterable<TKey, TValue> $iterable
+     * @param int|null               $max      Number of steps to complete the bar (0 if indeterminate), if null it will be inferred from $iterable
```
Comment (GromNaN) [inline src/Symfony/Component/Console/Helper/ProgressBar.php:312]: Could you copy this comment into `SymfonyStyle::progressIterate()`. `null`/`0` meanings needs to be documented.

Comment (jdreesen) [reply src/Symfony/Component/Console/Helper/ProgressBar.php:312]: Sure, done.

**Comment by jdreesen** [comment]
I think the failing test is unrelated.

**Review by javiereguiluz — APPROVED** [review]

**Review by GromNaN — APPROVED** [review]

**Comment by lyrixx** [comment]
Thank you @jdreesen.

**Pull request merged**